server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -1888,6 +1889,9 @@ public void flush(boolean force, boolean waitIfOngoing) throws EngineException {
                         }
 
                         translogManager.trimUnreferencedReaders();
+                    } catch (TranslogUploadFailedException e) {
+                        // Do not fail engine as this is due to translog upload failure
+                        throw new FlushFailedEngineException(shardId, e);
```

Comment:
Without this change as well, it would have thrown `FlushFailedEngineException` and should not have failed engine .